File: src/main/java/com/wxl/dyttcrawler/urlhandler/DyttUrlFilter.java
package com.wxl.dyttcrawler.urlhandler;

import com.wxl.dyttcrawler.core.DyttConstants;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

import java.net.URL;

/**
 * Create by wuxingle on 2020/5/16
 * url匹配
 * 协议，域名必须匹配
 */
@Slf4j
@Component
public class DyttUrlFilter implements UrlFilter {

    /**
     * 是否匹配
     */
    @Override
    public boolean filter(URL url) {
        String protocol = url.getProtocol();
        if (!DyttConstants.PROTOCOL_PATTERN.matcher(protocol).matches()) {
            return false;
        }
        String host = url.getHost();
        if (!DyttConstants.DOMAIN_PATTERN.matcher(host).matches()) {
            return false;
        }

        return true;
    }
}

File: src/main/kotlin/com/wxl/dyttcrawler/urlhandler/DyttUrlFilter.kt
package com.wxl.dyttcrawler.urlhandler

import com.wxl.dyttcrawler.properties.CrawlerProperties
import org.springframework.stereotype.Component
import java.net.URL
import java.util.regex.Pattern

/**
 * Create by wuxingle on 2021/10/12
 * url匹配
 * 协议，域名必须匹配
 */
@Component
class DyttUrlFilter(
    private val crawlerProperties: CrawlerProperties
) : UrlFilter {
    /**
     * 爬取协议正则
     */
    private val protocolPattern: Pattern = Pattern.compile("http|https")

    /**
     * 允许访问的域名
     */
    private var allowDomains: Set<String> = crawlerProperties.allowDomains.toSet()

    /**
     * url协议和域名必须匹配
     */
    override fun match(url: URL): Boolean =
        protocolPattern.matcher(url.protocol).matches()
                && allowDomains.contains(url.host)

}